src/libponyc/ast/error.c

```diff
@@ -131,15 +131,16 @@ static void errors_push(errors_t* errors, errormsg_t* e)
   if(errors->immediate_report)
     error_print(e, errors->output_stream);
 
-  if(errors->head == NULL)
+  if(errors->head != NULL) {
+    errors->tail->next = e;
+  }
+  else
   {
     errors->head = e;
-    errors->tail = e;
-  } else {
-    errors->tail->next = e;
-    errors->tail = e;
   }
 
+  errors->tail = e;
```

Comment:
i suppose you can make an argument that this removes a bit of duplication, but I don't think it aids in understanding. The existing duplication makes it very easy to see what is going on with updating tail.

Does anyone else have any opinions on moving the errors->tail = e to outside the if/else block?

Reply:
Okay, that's fair.